File: src/api/discussion-remote-mutate-draft.ts
import { Discussion, DiscussionCommand, type DiscussionDraftMessage, type SessionHandle } from "~/models";
import { discussionPostCommand } from "./private/discussion-post-command";
import { discussions } from "./discussions";
import { discussionMessages } from "./discussion-messages";

/**
 * Send local changes to the object
 * to the server.
 */
export const discussionRemoteMutateDraft = async (session: SessionHandle, discussion: Discussion, draft: DiscussionDraftMessage): Promise<void> => {
  await discussionPostCommand(session, DiscussionCommand.brouillon, {
    id: draft.possessionID,
    content: draft.content,
    replyMessageID: draft.replyMessageID
  });

  await discussions(session, discussion.cache);
  await discussionMessages(session, discussion);
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/discussion-remote-mutate-draft.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.api.private.discussionPostCommand
import ink.literate.pawnote.models.Discussion
import ink.literate.pawnote.models.DiscussionCommand
import ink.literate.pawnote.models.DiscussionDraftMessage
import ink.literate.pawnote.models.SessionHandle
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put

/** Send local changes to the object to the server. */
suspend fun discussionRemoteMutateDraft(
    session: SessionHandle,
    discussion: Discussion,
    draft: DiscussionDraftMessage
) {
  discussionPostCommand(
      session,
      DiscussionCommand.brouillon,
      buildJsonObject {
        put("id", draft.possessionID)
        put("content", draft.content)
        put("replyMessageID", draft.replyMessageID)
      })

  discussions(session, discussion.cache)
  discussionMessages(session, discussion)
}
